src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationLoaderSecurity7.java

```diff
@@ -101,6 +112,11 @@ public void success(SecurityDynamicConfiguration<?> dConf) {
                 if(log.isDebugEnabled()) {
                     log.debug("Received config for {} (of {}) with current latch value={}", dConf.getCType().toLCString(), Arrays.toString(events), latch.getCount());
                 }
+                // Audit configuration doc is available in the index.
+                // Configuration can be hot-reloaded.
+                if (dConf.getCType() == CType.AUDIT) {
+                    isAuditConfigDocPresentInIndex.set(true);
```

Comment:
`isAuditConfigDocPresentInIndex` must be set *before* `latch` is updated to avoid possibility of the waiting thread using not updated state.